File: AlgoExpert/2. Medium/C++/Move Element To End/Move Element To End.cpp
#include <iostream>
#include <vector>

using namespace std;

// O(n) time | O(1) space - where n is the length of the array

vector<int> moveElementToEnd(vector<int> array, int toMove) {
	int i = 0;
	int j = array.size() - 1;
	while (i < j) {
		while (i < j && array[j] == toMove)
			j--;
		if (array[i] == toMove)
			swap(array[i], array[j]);
		i++;
	}
	return array;
}

File: AlgoExpert/2. Medium/Python/Move Element To End/Move Element To End.py
# O(n) time | O(1) space - where n is the length of the array

def moveElementToEnd(array, toMove):
	i = 0
	j = len(array) - 1
	while i < j:
		while i < j and array[j] == toMove:
			j -= 1
		if array[i] == toMove:
			array[i], array[j] = array[j], array[i]
		i += 1
	return array